### PR TITLE
Fix deadlock in get_lock and expire_trees by:

### PR DESCRIPTION
**Pull request by JeetKunDoug**
- Making riak_kv_entropy_manager:get_lock have a defined timeout
- Making riak_kv_index_hashtree:expire a cast as that's how it's used.
  Relates to basho/yokozuna#644

### CODE REVIEW

File: src/riak_kv_index_hashtree.erl [hunk 1]
```diff
@@ -1008,6 +1009,9 @@ get_all_locks(Type, Index, Pid) ->
             end;
         _ ->
             false
+    catch exit:{timeout,_} ->
```
Comment (zeeshanlakhani) [inline src/riak_kv_index_hashtree.erl:1013]: mentioned a similar pattern here: https://github.com/basho/yokozuna/pull/644#discussion_r64288091. No biggie, but something I was thinking.

File: src/riak_kv_entropy_manager.erl [hunk 2]
```diff
@@ -79,6 +80,7 @@
 
 -define(DEFAULT_CONCURRENCY, 2).
 -define(DEFAULT_BUILD_LIMIT, {1, 3600000}). %% Once per hour
+-define(KV_ENTROPY_LOCK_TIMEOUT, app_helper:get_env(riak_kv, anti_entropy_lock_timeout, 10000)).
```
Comment (zeeshanlakhani) [inline src/riak_kv_entropy_manager.erl]: same idea mentioned in https://github.com/basho/yokozuna/pull/644#discussion_r64286497. `10000` may be nicer as it's own default macro.

**Comment by zeeshanlakhani** [comment]
+1 19216cc

**Comment by JeetKunDoug** [comment]
@borshop merge

**Pull request merged**